test/sequential/sequencerStaking.cancel.test.ts

```diff
@@ -197,71 +197,85 @@ it("GIVEN a slashed sequencer, WHEN slashed it can not provide any update / canc
-    await createAnUpdateAndCancelIt(testUser1, testUser2Address, chain);
+
+  const {
+    txIndex,
+    api: api1,
+    reqIdCanceled: reqIdCanceled1,
+  } = await createAnUpdateAndCancelIt(
+    testUser1,
+    testUser2.keyRingPair.address,
+    chain,
+  );
   await waitForNBlocks(disputePeriodLength);
-  const txIndex1 = await Rolldown.lastProcessedRequestOnL2(chain);
-  await Rolldown.waitForReadRights(judge.keyRingPair.address);
-  cancelResolutionEvents = await Sudo.asSudoFinalized(
+  const updateValue = new L2Update(api)
+    .withDeposit(
+      txIndex,
+      testUser2.keyRingPair.address,
+      testUser1.keyRingPair.address,
+      BN_MILLION,
+    )
+    .withDeposit(
+      txIndex + 1,
+      testUser2.keyRingPair.address,
+      testUser1.keyRingPair.address,
+      BN_MILLION,
+    )
+    .on(chain)
+    .build();
+  const { reqIdCanceled: reqIdCanceled2, api: api2 } =
+    await createAnUpdateAndCancelIt(
+      testUser2,
+      testUser1.keyRingPair.address,
+      chain,
+      updateValue,
+    );
+  await waitForNBlocks(disputePeriodLength);
+  await testUser2.refreshAmounts(AssetWallet.BEFORE);
+  const cancelResolutionEvent1 = await Sudo.asSudoFinalized(
     Sudo.sudoAsWithAddressString(
       judge.keyRingPair.address,
       new L2Update(api1)
-        .withCancelResolution(txIndex1, reqIdCanceled1, false)
+        .withCancelResolution(txIndex, reqIdCanceled1, false)
         .on("Ethereum")
         .build(),
     ),
   );
-  await waitSudoOperationSuccess(cancelResolutionEvents, "SudoAsDone");
+  await waitSudoOperationSuccess(cancelResolutionEvent1, "SudoAsDone");
   await waitForNBlocks(disputePeriodLength);
   await testUser2.refreshAmounts(AssetWallet.AFTER);
-  await signTx(
-    await getApi(),
-    await SequencerStaking.provideSequencerStaking(
-      (await SequencerStaking.minimalStakeAmount()).addn(1000),
-      "Ethereum",
-    ),
-    testUser2.keyRingPair,
-  ).then((events) => {
-    expectExtrinsicSucceed(events);
-  });
-  sequencers = await SequencerStaking.activeSequencers();
-  expect(sequencers.toHuman().Ethereum).toContain(judge.keyRingPair.address);
-  await testUser1.refreshAmounts(AssetWallet.BEFORE);
-  const { reqIdCanceled: reqIdCanceled2, api: api2 } =
-    await createAnUpdateAndCancelIt(
-      testUser2,
-      testUser1.keyRingPair.address,
-      chain,
-    );
-  await waitForNBlocks(disputePeriodLength);
-  const txIndex2 = await Rolldown.lastProcessedRequestOnL2(chain);
   await Rolldown.waitForReadRights(judge.keyRingPair.address);
-  cancelResolutionEvents = await Sudo.asSudoFinalized(
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  const cancelResolutionEvent2 = await Sudo.asSudoFinalized(
     Sudo.sudoAsWithAddressString(
       judge.keyRingPair.address,
       new L2Update(api2)
-        .withCancelResolution(txIndex2, reqIdCanceled2, false)
+        .withCancelResolution(txIndex + 1, reqIdCanceled2, false)
```

Comment:
here I think you can use the getLatestExecutedRequestId, but I think its ok with those txindex+1

Reply:
I changed it